frontend/src/components/ProfileContent.tsx

```diff
@@ -50,64 +61,199 @@ const useStyles = makeStyles({
         padding: 5,
         paddingRight: 10,
     },
-});
+    bottomRight: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-10%, -10%)',
+    },
+    bottomRight2: {
+        position: 'absolute',
+        bottom: '0',
+        right: '0',
+        transform: 'translate(-100%, -10%)',
+    },
+})
 
 export default function ProfileContent(props: ProfileContentProps) {
-    const classes = useStyles();
+    const classes = useStyles()
+    const [isEditMode, toggleEditMode] = React.useState(false)
+
+    // Custom profile hook. This separates saved profile content from edited, unsaved profile content.
+    function useProfile(initialProfile: ProfileContentProps) {
+        const [name, setName] = React.useState<typeof initialProfile.name>(initialProfile.name)
+        const [username, setUsername] = React.useState<typeof initialProfile.username>(initialProfile.username)
+        const [age, setAge] = React.useState<typeof initialProfile.age>(initialProfile.age)
+        const [gender, setGender] = React.useState<typeof initialProfile.gender>(initialProfile.gender)
+        const [occupations, setOccupations] = React.useState<typeof initialProfile.occupations>(initialProfile.occupations)
+        const [location, setLocation] = React.useState<typeof initialProfile.location>(initialProfile.location)
+        const [interests, setInterests] = React.useState<typeof initialProfile.interests>(initialProfile.interests)
+        const [religion, setReligion] = React.useState<typeof initialProfile.religion>(initialProfile.religion)
+        const [profileImg, setProfileImg] = React.useState<typeof initialProfile.profileImg>(initialProfile.profileImg)
+
+        const editableContent: ProfileContentProps = {
+            name: name,
+            username: username,
+            age: age,
+            gender: gender,
+            occupations: occupations,
+            location: location,
+            interests: interests,
+            religion: religion,
+            profileImg: profileImg,
+        }
+
+        const setters = {
+            setName,
+            setUsername,
+            setAge,
+            setGender,
+            setOccupations,
+            setLocation,
+            setInterests,
+            setReligion,
+            setProfileImg,
+        }
+
+        return { editableContent, setters }
+    }
 
-    // Add user profile values to list.
-    var profileFields: ProfileField[] = [
-        {
+    // Pass props to to useProfile hook.
+    const { editableContent, setters } = useProfile(props)
+
+    // Initialize readonly profile content and acquire hook to update it when edits are saved.
+    const [readonlyContent, setProfile] = React.useState<ProfileContentProps>(editableContent);
```

Comment:
@Deravadom is this what you had in mind? it seems to work very well

Reply:
Exactly

Reply:
Although I'd personally pull it out of `src/components/ProfileContent.tsx` and into `src/utils/profile.tsx` or something similar.